src/main/scala/org/ergoplatform/nodeView/wallet/WalletStorage.scala

```diff
@@ -51,14 +51,14 @@ class WalletStorage extends ScorexLogging {
 
   private def put(trackedBox: TrackedBox): Option[TrackedBox] = {
     if (trackedBox.certainty == Uncertain) uncertainBoxes += trackedBox.boxId
-    if (trackedBox.spendingStatus == Unspent) unspentBoxes += trackedBox.boxId
+    if (trackedBox.spendingStatus == Unspent && trackedBox.certainty == Certain) unspentBoxes += trackedBox.boxId
     registry.put(trackedBox.boxId, trackedBox)
   }
 
   private def remove(boxId: ModifierId): Option[TrackedBox] = {
```

Comment:
Looks like all we need here is 
```
    uncertainBoxes.remove(boxId)
    unspentBoxes.remove(boxId)
    registry.remove(boxId)
```

Reply:
@catena2w, why making potentially redundant function calls?